variables.tf

```diff
@@ -0,0 +1,224 @@
+variable "public_ip_config" {
+  type = object({
+    enabled           = bool
+    allocation_method = optional(string, "Static")
+  })
+  default = {
+    enabled = false
+  }
+  validation {
+    condition     = contains(["Static", "Dynamic"], var.public_ip_config.allocation_method)
+    error_message = "Allocation method must be Static or Dynamic"
+  }
+  description = <<-DOC
+  ```
+    enabled: Optionally select true if a public ip should be created. Defaults to false.
+    allocation_method: The allocation method of the public ip that will be created. Defaults to static.      
+  ```
+  DOC
+}
+
+# nsg needs to be an object to use the count object in main.tf. 
+variable "nic_config" {
+  type = object({
+    private_ip                    = optional(string)
+    dns_servers                   = optional(list(string))
+    enable_accelerated_networking = optional(bool, false)
+    nsg = optional(object({
+      id = string
+    }))
+  })
+  default     = {}
+  description = <<-DOC
+  ```
+    private_ip: Optioanlly specify a private ip to use. Otherwise it will  be allocated dynamically.
+    dns_servers: Optionally specify a list of dns servers for the nic.
+    enable_accelerated_networking: Enabled Accelerated networking (SR-IOV) on the NIC. The machine SKU must support this feature.
+    nsg: Although it is discouraged you can optionally assign an NSG to the NIC. Optionally specify a NSG object.
+  ```
+  DOC
+}
+
+variable "additional_network_interface_ids" {
+  type        = list(string)
+  default     = []
+  description = "List of ids for additional azurerm_network_interface."
+}
+
+variable "subnet" {
+  type = object({
+    id               = string
+    address_prefixes = list(string)
+  })
+  description = "The variable takes the subnet as input and takes the id and the address prefix for further configuration."
+}
+
+variable "admin_username" {
+  type        = string
+  description = "Optionally choose the admin_username of the vm. Defaults to loc_sysadmin."
+  default     = "loc_sysadmin"
+}
+
+variable "admin_credential" {
+  type = object({
+    admin_password = optional(string)
+    public_key     = optional(string)
+  })
+
+  validation {
+    condition     = (var.admin_credential.admin_password != null && var.admin_credential.public_key == null) || (var.admin_credential.admin_password == null && var.admin_credential.public_key != null)
+    error_message = "Use admin password or public ssh key."
+  }
+  sensitive   = true
+  description = <<-DOC
+  ```
+    admin_password: Password of the local administrator.
+    public_key: SSH public key file (e.g. file(id_rsa.pub))
+  ```
```

Comment:
How do I know, that I can only specify exactly one?

Reply:
The validation will tell you. I will add it to the description too